### PR TITLE
enforce action mobx

### PR DESCRIPTION
**Pull request by shayanbinary**

### CODE REVIEW

File: src/store/ShareStore.js [hunk 1]
```diff
@@ -67,19 +64,23 @@ export default class ShareStore {
         hashPromise
             .then((hash) => {
```
Comment (brucebinary) [inline src/store/ShareStore.js]: How about...

```js
hasPromise
    .then(this.onHashAvailable)
    .catch(this.onHashError);
```

I'm not sure whether you can use [`runInAction`](https://mobx.js.org/refguide/action.html#-runinaction-name-thunk) but I think this is pretty readable to me.

File: src/store/TimeperiodStore.js [hunk 2]
```diff
@@ -78,16 +58,36 @@ export default class TimeperiodStore {
...
             }
 
-            if (callFromSettings) { setRemain(); }
+            if (callFromSettings) { this.setRemain(); }
 
             if (!this.countdownInterval) {
                 this.countdownInterval = setInterval(() => {
-                    setRemain();
+                    this.setRemain();
                 }, 1000);
```
Comment (brucebinary) [inline src/store/TimeperiodStore.js]: not exactly related to this PR, but now that I do notice it...

`this.remain` should not be an `@obervable`.

File: src/store/ShareStore.js (hunk 1 above)
Comment (shayanbinary) [reply src/store/ShareStore.js]: @brucebinary amazing solution

File: src/store/TimeperiodStore.js (hunk 2 above)
Comment (shayanbinary) [reply src/store/TimeperiodStore.js]: @brucebinary  I checkout the changes for this file and just remove `@obervable` from `remain`

File: src/store/ShareStore.js [hunk 3]
```diff
@@ -39,14 +39,10 @@ export default class ShareStore {
...
-
     @observable shareLink = '';
 
-    fixedEncodeURIComponent(str) {
-        return encodeURIComponent(str).replace(/[!'()*]/g, c => `%${c.charCodeAt(0).toString(16)}`);
-    }
-    refereshShareLink = () => {
-        const self = this;
+
+    @action refereshShareLink = () => {
```
Comment (brucebinary) [inline src/store/ShareStore.js]: shouldn't this be `@action.bound refreshShareLink(str) {`?

also do fix the typo. Aside that this looks good to me

File: src/store/ShareStore.js [hunk 4]
```diff
@@ -42,7 +42,7 @@ export default class ShareStore {
     @observable shareLink = '';
 
 
-    @action refereshShareLink = () => {
+    @action.bound refereshShareLink() {
```
Comment (brucebinary) [inline src/store/ShareStore.js]: Bro. It's "refresh" not "referesh".

**Pull request merged**